### ISSUE TITLE
Dollar signs not escaped in generated toString()

### ISSUE THREAD

**Issue by terrier989** [google/built_value.dart#784]
Classes like:
```
class Example {
  String get $notEscaped;
}
```

Generate:
```
  @override
  String toString() {
    return (newBuiltValueToStringHelper('Example')
          ..add('$notEscaped', notEscaped))
        .toString();
 }
```

The bug is in [value_source_class.dart](https://github.com/google/built_value.dart/blob/master/built_value_generator/lib/src/value_source_class.dart):
```
    if (!implementsToString) {
      result.writeln('@override');
      result.writeln('String toString() {');
      if (fields.isEmpty) {
        result
            .writeln("return newBuiltValueToStringHelper('$name').toString();");
      } else {
        result.writeln("return (newBuiltValueToStringHelper('$name')");
        result.writeln(fields
            .map((field) => "..add('${field.name}',  ${field.name})")
            .join(''));
        result.writeln(').toString();');
      }
      result.writeln('}');
      result.writeln();
    }
```

**Comment by davidmorgan** [comment]
Thanks.

Can I ask why you need this? Dollars are not usually in field names.

If it's for serialization, you can instead use a standard name and use the `@BuiltValueField` annotation to set the wire name.

Thanks :)

**Comment by terrier989** [comment]
I considered a design that separates metadata properties from ordinary properties with a dollar prefix (just for clarity).

**Comment by davidmorgan** [comment]
Fixed in #804 -- thanks :)
